packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -705,42 +721,64 @@ export class GarbageCollector implements IGarbageCollector {
-            ...this.gcOptions,
-        });
         if (this.isSummarizerClient) {
             this.mc.logger.sendTelemetryEvent({
                 eventName: "GarbageCollectorLoaded",
-                gcConfigs: gcConfigProps,
+                gcConfigs: JSON.stringify({
+                    gcEnabled: this.gcEnabled,
+                    sweepEnabled: this.sweepEnabled,
+                    runGC: this.shouldRunGC,
+                    runSweep: this.shouldRunSweep,
+                    writeAtRoot: this._writeDataAtRoot,
+                    testMode: this.testMode,
+                    sessionExpiry: this.sessionExpiryTimeoutMs,
+                    inactiveTimeout: this.inactiveTimeoutMs,
+                    existing: createParams.existing,
+                    trackGCState: this.trackGCState,
+                    ...this.gcOptions,
+                }),
             });
         }
+    }
 
-        // Initialize the base state that is used to detect when inactive objects are used.
-        if (this.shouldRunGC) {
+    public setConnectionState(connected: boolean, clientId?: string | undefined): void {
+        /**
+         * For non-summarizer clients, initialize the base state when the container becomes active, i.e., it transitions
+         * to "write" mode. This will ensure that the container's own join op is processed and there is a recent
+         * reference timestamp that will be used to update the state of unreferenced nodes. Also, all trailing ops which
+         * could affect the GC state will have been processed.
+         *
+         * Ideally, this initialization should only be done for summarizer client. However, we are currently rolling out
+         * sweep in phases and we want to track when inactive and sweep ready objects are used in any client.
+         */
+        if (this.activeConnection() && !this.isSummarizerClient && this.shouldRunGC) {
             this.initializeBaseStateP.catch((error) => {
                 const dpe = DataProcessingError.wrapIfUnrecognized(
                     error,
                     "FailedToInitializeGC",
                 );
-                dpe.addTelemetryProperties({ gcConfigs: gcConfigProps });
+                dpe.addTelemetryProperties({
+                    gcConfigs: JSON.stringify({
+                        gcEnabled: this.gcEnabled,
+                        sweepEnabled: this.sweepEnabled,
+                        runGC: this.shouldRunGC,
+                        runSweep: this.shouldRunSweep,
+                        writeAtRoot: this._writeDataAtRoot,
+                        testMode: this.testMode,
+                        sessionExpiry: this.sessionExpiryTimeoutMs,
+                        inactiveTimeout: this.inactiveTimeoutMs,
+                        trackGCState: this.trackGCState,
+                        ...this.gcOptions,
+                    }),
+                });
                 throw dpe;
             });
         }
```

Comment:
What happens when a client disconnects after this.initializeBaseStateP has been called and completed?

Reply:
I am not sure what you mean. What happens to ?

Reply:
So lets say a summarizer client disconnects for a second or two and the connectionState is changed from disconnected to connected. Are there any issues afterwards when GC runs, or while it runs?

Reply:
this.initliazeBaseStateP is a lazy promise. The first time the client connects in write mode, it will be called and will initialize the state. The base state does not change after that so any disconnect / reconnect will not matter.